llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1456,6 +1476,107 @@ void AsmPrinter::emitBBAddrMapSection(const MachineFunction &MF) {
         OutContext.reportError(SMLoc(), "Couldn't find the block's index");
       OutStreamer->emitULEB128IntValue(I);
     }
+
+    // Emit call marker table for the Yk JIT.
+    //
+    // The table is a size header, followed by call instruction addresses.
+    //
+    // YKOPT: to save space, instead of using absolute symbol addresses, compute
+    // the distance from the start of the block and use uleb128 encoding.
+    const size_t NumCalls = YkCallMarkerSyms[&MBB].size();
+    OutStreamer->emitULEB128IntValue(NumCalls);
+    for (auto Tup : YkCallMarkerSyms[&MBB]) {
+      // Emit address of the call instruction.
+      OutStreamer->emitSymbolValue(std::get<0>(Tup), getPointerSize());
+      // Emit address of target if known, or 0.
+      MCSymbol *Target = std::get<1>(Tup);
+      if (Target)
+        OutStreamer->emitSymbolValue(Target, getPointerSize());
+      else
+        OutStreamer->emitIntValue(0, getPointerSize());
+    }
+
+    // Emit successor information.
+    //
+    // Each codegenned block gets a record indicating any statically known
+    // successors. The record starts with a `SuccessorKind` byte:
+    enum SuccessorKind {
+      // One successor.
+      Unconditional = 0,
+      // Choice of two successors.
+      Conditional = 1,
+      // A control flow edge known only at runtime, e.g. a return or an indirect
+      // branch.
+      Dynamic = 2,
+    };
+
+    // Then depending of the `SuccessorKind` there is a payload describing the
+    // successors:
+    //
+    // `Unconditional`: [target-address: u64]
+    // `Conditional`:   [taken-address: u64, not-taken-address: u64]
+    // `Dynamic`:       [] (i.e. nothing, becuase nothing is statically known)
```

Comment:
s/becuase/because

Reply:
Thanks. Fixed in [26f349d](https://github.com/ykjit/ykllvm/pull/50/commits/26f349d237ce634cca96402edee55e6ab82e33a5)